projects/core/components/dialog/dialog.component.ts

```diff
@@ -70,7 +90,7 @@ export class TuiDialogComponent<O, I> {
             : this.animation;
     }
 
-    close(): void {
+    private close(): void {
```

Comment:
It's breaking, what do you think about

```ts
close(): void {
  this.close$.next();
}

private immediatelyClose(): void {
  if (this.context.required) {
      this.context.$implicit.error(REQUIRED_ERROR);
  } else {
  // ...
}
```

Reply:
Public code don't really have access to this class.